src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -14,7 +15,11 @@ internal abstract partial class BoundExpression
         /// </summary>
         public virtual object Display
         {
-            get { return this.Type; }
+            get
+            {
+                Debug.Assert((object)this.Type != null, $"Unexpected null type in {this.GetType().Name}");
```

Comment:
This returns the type whenever the type is non-null. The type can only be null for certain kinds of nodes, and all of those nodes have their own override of the `Display` property.